src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java

```diff
@@ -208,6 +191,46 @@ private void handleClientListBySite(RoutingContext rc) {
         }
     }
 
+    private void handleClientByKeyId(RoutingContext rc) {
+        try {
+            String keyId;
+
+            try{
+                keyId = rc.queryParam("keyId").get(0);
+            } catch (Exception e) {
+                ResponseUtil.error(rc, 400, "unable to parse keyId " + e.getMessage());
+                return;
+            }
+
+            LegacyClientKey clientKey = this.clientKeyProvider.getAll().stream().filter(legacyClientKey -> legacyClientKey.getKeyId().equals(keyId)).findFirst().orElse(null);
```

Comment:
I know it's only a one-liner, but it's a pretty long line. I wonder if it is worth extracting to:
```
private LegacyClientKey getClientKey(String keyId) {
    return this.clientKeyProvider.getAll()
            .stream()
            .filter(legacyClientKey -> legacyClientKey.getKeyId().equals(keyId))
            .findFirst()
            .orElse(null);
}
```

Reply:
Means we can better separate concerns - all I care about at this level of abstraction is that I'm getting a client key based off the `keyId`

Reply:
Yeah that makes sense!